src/hotspot/share/gc/parallel/psCompactionManager.cpp

```diff
@@ -92,7 +92,7 @@ void ParCompactionManager::initialize(ParMarkBitMap* mbm) {
 
   _shadow_region_array = new (ResourceObj::C_HEAP, mtGC) GrowableArray<size_t >(10, mtGC);
 
-  _shadow_region_monitor = new Monitor(Mutex::barrier, "CompactionManager monitor",
+  _shadow_region_monitor = new Monitor(Mutex::nosafepoint, "CompactionManager_lock",
```

Comment:
Not clear why this one needed to change??

Reply:
This one changes because 'barrier'  is above 'leaf' which checks for safepoint.  nosafepoint it the top rank that doesn't check for safepoint, so this was made nosafepoint.

Reply:
Okay